dags/synapse-datasets-created-7-days.py

```diff
@@ -0,0 +1,221 @@
+                entity_created_date >= DATEADD(DAY, -7, '{query_date or context["params"]["current_date_time"]}')
+            )
+        ORDER BY entity_created_date;
+        """
+        print(query)
+        cs.execute(query)
+        created_entity_df = cs.fetch_pandas_all()
+
+        entity_created = []
+        for _, row in created_entity_df.iterrows():
+            entity_created.append(
+                EntityCreated(
+                    name=row["NAME"],
+                    id=row["ID"],
+                    node_type=row["NODE_TYPE"],
+                    parent_id=row["PARENT_ID"],
+                    content_type=row["CONTENT_TYPE"],
+                    created_on=row["ENTITY_CREATED_DATE"],
+                    user_name=row["USER_NAME"],
+                    created_by=row["CREATED_BY"],
+                    is_public=row["IS_PUBLIC"],
+                )
+            )
+        return entity_created
+
+    @task
+    def generate_slack_message(entity_created: List[EntityCreated], **context) -> str:
+        """Generate the message to be posted to the slack channel."""
+        message = ":synapse: Datasets or projects created in the last 7 days \n\n"
+        for index, row in enumerate(entity_created):
+            if row.content_type:
+                data_type = row.content_type
+            else:
+                data_type = row.node_type
+            message += f"{index+1}. <https://www.synapse.org/#!Synapse:syn{row.id}|*{row.name}*> (Type: {data_type}, Created on: {row.created_on}, Created by: <https://www.synapse.org/Profile:{row.created_by}/profile|{row.user_name}>, Public: {row.is_public})\n\n"
+        return message
+
+    @task
+    def post_slack_messages(message: str) -> bool:
+        """Post the top downloads to the slack channel."""
+        client = WebClient(token=Variable.get("SLACK_DPE_TEAM_BOT_TOKEN"))
+        result = client.chat_postMessage(channel="hotdrops", text=message)
+        print(f"Result of posting to slack: [{result}]")
+        return result is not None
+
+    @task
+    def push_results_to_synapse_table(
+        entity_created: List[EntityCreated], **context
+    ) -> None:
+        """Push the results to a Synapse table."""
+        data = []
+        # convert context["params"]["backfill_date_time"] to date in same format as date.today()
+        today = (
+            date.today()
+            if not context["params"]["backfill"]
+            else datetime.strptime(
+                context["params"]["backfill_date_time"], "%Y-%m-%d  %H:%M:%S"
+            ).date()
+        )
+        for row in entity_created:
+            data.append(
+                [
+                    row.name,
+                    row.id,
+                    row.node_type,
+                    row.parent_id,
+                    row.content_type,
+                    row.created_on,
+                    row.created_by,
+                    row.is_public,
+                    today,
+                ]
```

Comment:
Should the Synapse table include all attributes from the `EntityCreated`? Totally fine if not, but if it should, the first name/last name is missing.

Reply:
I added full_name so that the slack message would be easier to read. But I don't think it matters for the synapse table?

Reply:
It depends on what the requirements of the ticket are

Reply:
Okay let me ask that in the Jira ticket.

Reply:
Have gotten the clarification from Tom, and he said that we don't need to include that in the synapse table. I will mark this as "resolved" then.